spinalcordtoolbox/reports/assets/_assets/js/main.js

```diff
@@ -147,35 +147,64 @@ $(document).ready(function(){
-          ? heavy_excl_mark
-          : sct_data[index].qc === heavy_excl_mark
-          ? empty_state
-          : heavy_check_mark
-      );
-      // Save QC state to local storage
-      var uniqueId = sct_data[index].moddate + '_' + sct_data[index].fname_in + '_' + sct_data[index].command;
-      localStorage.setItem(uniqueId, sct_data[index].qc);
-      // Update table display with updated sct_data
-      set_download_yml_btn_state(heavy_excl_mark);
-      set_download_yml_btn_state(heavy_ballot_x);
-      $("#table").bootstrapTable({data: sct_data});
-      $("#table").bootstrapTable("load", sct_data);
-      hideColumns();
-      // Set focus on the row that was just updated
-      document.getElementById("table").rows[0].classList.remove("active");
-      document.getElementById("table").rows[parseInt(rel_index)+1].className="active";
-      selected_row = document.getElementById("table").rows[parseInt(rel_index)+1].innerText;
-      document.getElementById("table").rows[parseInt(rel_index)+1].scrollIntoViewIfNeeded(false);
+    // Keys that update table state (f key, number keys)
+    if ((evt.which == 70) || (evt.which >= 48 && evt.which <= 57) || (evt.which >= 96 && evt.which <= 105)) {
+      // Only attempt to update the cell value if the object is a row (i.e. if it has length)
+      if (obj.length > 0) {
+        // Fetch the index into `sct_data` corresponding to the selected row
+        var cols = getActiveColumns();
+        var vals = obj[0].innerText.split("\t");
+        let rel_index = obj[obj.length - 1].getAttribute("data-index");
+        let index = sct_data.findIndex(y => check_element(y,cols,vals))
+        var uniqueId = sct_data[index].moddate + '_' + sct_data[index].fname_in + '_' + sct_data[index].command;
+
+        // Update the cell value in `sct_data` based on the key pressed
+        if (evt.which == 70) {
+          // f key => cycle through qc states
+          sct_data[index].qc = (
+            sct_data[index].qc === heavy_check_mark
+            ? heavy_ballot_x
+            : sct_data[index].qc === heavy_ballot_x
+            ? heavy_excl_mark
+            : sct_data[index].qc === heavy_excl_mark
+            ? empty_state
+            : heavy_check_mark
+          );
+          // Save QC state to local storage
+          localStorage.setItem(uniqueId+"_qc", sct_data[index].qc);
+          // Update table display with updated sct_data
+          set_download_yml_btn_state(heavy_excl_mark);
+          set_download_yml_btn_state(heavy_ballot_x);
+
+        }
+        if ((evt.which >= 48 && evt.which <= 57) || (evt.which >= 96 && evt.which <= 105)) {
+          // Normalize numpad event codes (to avoid writing non-numeric characters to the table)
+          let code = evt.which;
+          if (evt.which >= 96 && evt.which <= 105) {
+            code -= 48;
+          }
+          // 0 key, store "None"
+          if (code == 48) {
+            sct_data[index].rank = "None"
```

Comment:
We should probably store an empty string, rather than `None`, to match the behaviour of an untouched column (both visually in the QC, and in the JSON file generated by "Save All").
```suggestion
            sct_data[index].rank = ""
```

Reply:
This makes sense to me, and I like the suggestion, but as @plbenveniste is requesting this feature specifically for his own research, and has requested "None" specifically, I'm not sure if there is a reason for that text that he needs?

I think maybe it could be reasonable to merge this PR as-is, let @plbenveniste try it out for his data, while keeping a specific eye on the None/"" fields, and if there is a need, maybe change it in a follow-up PR?

Alternatively we could add event handling for the backspace/delete key to "clear" the row (thus gaining the same "" behavior while still preserving "none"). But I wouldn't want to hold up this PR any further if it is urgent.

Reply:
about the None vs. empty: I think the behaviour should be the same as for the check/fail/warning column, ie: there is no "None", but there is the possibility to go back to the empty value (eg: if we click on 'check' accidentallly and we want to go back to the pristine state)

Reply:
I agree with the last solution. I like the idea of being able to go back to an empty entry.